pandas/_libs/tslibs/strptime.pyx

```diff
@@ -60,15 +64,33 @@ cdef bint format_is_iso(f: str):
     Generally of form YYYY-MM-DDTHH:MM:SS - date separator can be different
     but must be consistent.  Leading 0s in dates and times are optional.
     """
-    excluded_formats = ["%Y%m"]
-
-    for date_sep in [" ", "/", "\\", "-", ".", ""]:
-        for time_sep in [" ", "T"]:
-            for micro_or_tz in ["", "%z", ".%f", ".%f%z"]:
-                iso_fmt = f"%Y{date_sep}%m{date_sep}%d{time_sep}%H:%M:%S{micro_or_tz}"
-                if iso_fmt.startswith(f) and f not in excluded_formats:
-                    return True
-    return False
+    iso_regex = re.compile(
+        r"""
+        ^                      # start of string
+        (?:                    # group for alternative date formats
+        %Y[ -/\\.]%m[ -/\\.]%d # 4-digit year, 2-digit month, 2-digit day
```

Comment:
`date_sep` needs to be the same, e.g. `%Y-%m/%d` isn't allowed, but `%Y-%m-%d` and `%Y/%m/%d` is

Reply:
sorry, didn't get, what you mean but, 

- **%Y[ -/\\.]%m[ -/\\.]%d**: 4-d-year 2d-month & 2-d-day separated by (space, a hyphen, forward slash, or backward slash)
- **%Y%m%d**: same as above without the separators 
- **%Y[ -/\\.]%m**: 4-d-year & 2-d-month with separated by (space, a hyphen, forward slash, or backward slash)
- **%Y%m**: same without separators
- **%Y**: 4-digit year.

Reply:
check the non-regex code, `date_sep` needs to be the same between %Y and %m and between %m and %d

`format_is_iso('%Y-%m/%d')` should return `False`